### PR TITLE
Fix docker images rebuilt issue when building each host image

### PR DESCRIPTION
**Pull request by xumia**
**- Why I did it**
The docker images rebuilt when building each host image in nightly build, it is not as expect.
It is caused by some of source files touched during building images, such as applying patches to source folders.

**- How I did it**
Before the .dep.sha files generated, check if the hash values changed, if the content is exactly the same as the old one, not necessary to touch the it.

**- How to verify it**
make SONIC_DPKG_CACHE_METHOD=wcache SONIC_DPKG_CACHE_SOURCE=/cache target/sonic-broadcom.bin
make SONIC_DPKG_CACHE_METHOD=wcache SONIC_DPKG_CACHE_SOURCE=/cache target/sonic-aboot-broadcom.swi
No docker images rebuilt

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [ ] 201911
- [ ] 202006

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by xumia** [comment]
retest mellanox please

**Review by yxieca — APPROVED** [review]

File: platform/broadcom/sai.dep [hunk 1]
```diff
@@ -5,7 +5,7 @@ DEP_FILES   := $(SONIC_COMMON_FILES_LIST) platform/broadcom/sai.mk platform/broa
 DEP_FILES   += $(SONIC_COMMON_BASE_FILES_LIST)
 # Get the Latest HTTP Header and calculate the SHA value as it is a softlink that always points to LATEST_INT_OCP_SAI_X.X.X
 SAI_FLAGS   := $(shell wget --spider --server-response $($(BRCM_SAI)_URL) $($(BRCM_SAI_DEV)_URL)  2>&1 \
-			   | grep -Ev -- '--|Date:'|sha256sum|awk '{print $$1}' )
+			   | grep -Ev -- '--|Date:'| grep -v 'x-ms-request-id' |sha256sum|awk '{print $$1}' )
```
Comment (qiluo-msft) [inline platform/broadcom/sai.dep:8]: >x-ms-request-id [](start = 41, length = 15)

You can combine it with previous regex, `|` means OR.

**Comment by qiluo-msft** [comment]
@Kalimuthu-Velappan Could you help review?

File: platform/broadcom/sai.dep (hunk 1 above)
Comment (Kalimuthu-Velappan) [reply platform/broadcom/sai.dep:8]: As Qi mentioned, you can try as bellow.

| grep -Ev -- '--|Date:|x-ms-request-id'|sha256sum|awk '{print $$1}'

**Comment by xumia** [comment]
retest broadcom please

**Review by qiluo-msft — APPROVED** [review]

File: Makefile.cache [hunk 2]
```diff
@@ -558,8 +558,10 @@ ALL_DEP_FILES_LIST += $(foreach pkg,$(2), $($(filter none,$($(1)_CACHE_MODE)), \
 $(addsuffix .$(3),$(addprefix $(2)/, $(1))) : $(2)/%.$(3) : \
 	$(2)/%.flags $$$$($$$$*_DEP_FILES) $$$$(if $$$$($$$$*_SMDEP_FILES), $(2)/%.smdep)
 	@$$(eval $$*_DEP_FILES_MODIFIED := $$? )
-	@$$(file >$$@,$$($$*_DEP_FILES))
-	@cat $$@ |xargs git hash-object >$$@.sha
+	@$$(file >$$@.tmp,$$($$*_DEP_FILES))
+	@cat $$@.tmp |xargs git hash-object >$$@.sha.tmp
+	@if ! cmp -s $$@.sha.tmp $$@.sha; then cp $$@.tmp $$@; cp $$@.sha.tmp $$@.sha; fi
+	@rm -f $$@.tmp $$@.sha.tmp
```
Comment (ouxlwhu) [inline Makefile.cache:564]: I find an issue that if I modify a file in a submodule, the corresponding packages could not be rebuilt. Eg, src/sonic-swss/cfgmgr/portmgr.cpp is modified，and make swss.deb will not be rebuilt, only swss.smdep and swss.smdep.smsha update, swss.dep is not touched, so target of swss.deb will not be executed.

**Pull request merged**